docs/source/tutorials/first_raw.md

```diff
@@ -0,0 +1,178 @@
+The script can be made executable and then run to execute the example calculation.
+:::
+
+
+
+First import the required classes and functions:
+
+```python
+from aiida.plugins import CalculationFactory
+from aiida import engine
+from aiida.orm import SinglefileData, load_code
+```
+
+Then, load the code that was setup in AiiDA for `lmp` and get an instance of the [process builder](https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/processes/usage.html#process-builder):
+
+```python
+# Load the code configured for ``lmp``. Make sure to replace
+# this string with the label used in the code setup.
+code = load_code('lammps@localhost')
+builder = CalculationFactory("lammps.raw").get_builder()
+builder.code = code
+```
+
+The process builder can be used to assign and automatically validate the inputs that will be used for the calculation.
+
+For the raw calculation the most important piece is to pass the LAMMPS script that will be run. To be able to pass it to AiiDA one needs to store it as a `SinglefileData` node, which basically stores a file in the AiiDA provenance graph. When a `LammpsRawCalculation` is submitted this file will be copied **exactly** in the machine performing the calculation.
+```python
+import io
+import textwrap
+
+script = SinglefileData(
+    io.StringIO(
+        textwrap.dedent(
+            """
+            # Rhodopsin model
+
+            units           real
+            neigh_modify    delay 5 every 1
+
+            atom_style      full
+            bond_style      harmonic
+            angle_style     charmm
+            dihedral_style  charmm
+            improper_style  harmonic
+            pair_style      lj/charmm/coul/long 8.0 10.0
+            pair_modify     mix arithmetic
+            kspace_style    pppm 1e-4
+
+            read_data       data.rhodo
+
+            fix             1 all shake 0.0001 5 0 m 1.0 a 232
+            fix             2 all npt temp 300.0 300.0 100.0 &
+                    z 0.0 0.0 1000.0 mtk no pchain 0 tchain 1
+
+            special_bonds   charmm
+
+            thermo          50
+            thermo_style    multi
+            timestep        2.0
+
+            run     100
+            """
+        )
+    )
+)
+builder.script = script
+```
+
+As one can notice the script wants to read a file named `data.rhodo` via the [`read_data`](https://docs.lammps.org/read_data.html) command. One can pass any set of files that the script might need, in this case a file stored in the lammps repository that is downloaded using the [requests library](https://docs.python-requests.org/en/latest/index.html), by first storing them as `SinglefileData` nodes and the passing them to the builder as follows:
+
+```python
+import requests
+data = SinglefileData(
+    io.StringIO(
+		textwrap.dedent(
+			requests.get(
+				"https://raw.githubusercontent.com/lammps/lammps/develop/bench/data.rhodo"
+			).text
+		)
+    )
+)
```

Comment:
The indentation was making it really difficult to read and understand
```suggestion
request = requests.get("https://raw.githubusercontent.com/lammps/lammps/develop/bench/data.rhodo")
data = SinglefileData(io.StringIO(request.text)))
```
By the way, in AiiDA v2.5 (to be released soon), we will be able to do `SinglefileData.from_string(request.text)`. No longer the need to import and use `io.StringIO` as the go between. Small usability improvement

Reply:
This is super nice! Looking forward to that feature!